scripts/prepdocslib/jsonparser.py

```diff
@@ -0,0 +1,21 @@
+import json
+from typing import IO
+
+from .page import Page
+
+
+class JsonParser:
+    """
+    Concrete parser that can parse json into pages
+    """
+
+    def parse(self, content: IO):
+        offset = 0
+        data = json.loads(content.read())
+        if isinstance(data, list):
+            for i, obj in enumerate(data):
+                page_text = json.dumps(obj)
+                offset += len(page_text)
```

Comment:
I need to improve this as offset is wrong for large files.

Reply:
Is this comment still the case?

Reply:
I just pushed a change to the offset calculation with some test asserts. I'm not sure if I'm missing a more complex issue with offset, let me know if you were thinking of another issue around offset.